### PR TITLE
Switch from Atlas Rest to Atlas Kafka

### PR DESCRIPTION
**Pull request by arempter**

### CODE REVIEW

File: src/main/scala/com/ing/wbaa/airlock/proxy/provider/LineageProviderAtlas.scala [hunk 1]
```diff
@@ -1,65 +1,70 @@
...
         // put object - copy
         // if contains header x-amz-copy-source
-        case HttpMethods.PUT if lineageHeaders.copySource.getOrElse("").length > 0 => lineageForCopyOperation(lineageHeaders, userSTS, Write(), clientIPAddress)
+        case HttpMethods.PUT if lineageHeaders.copySource.getOrElse("").length > 0 && !bucketObject.isEmpty =>
+          lineageForCopyOperation(lineageHeaders, userSTS, Write(), clientIPAddress)
 
         // post object (complete multipart)
         // aws request eg. POST /ObjectName?uploadId=UploadId and content-type application/xml
-        case HttpMethods.POST if lineageHeaders.queryParams.getOrElse("").contains("uploadId") => readOrWriteLineage(lineageHeaders, userSTS, Write(), clientIPAddress)
+        case HttpMethods.POST if lineageHeaders.queryParams.getOrElse("").contains("uploadId") && !bucketObject.isEmpty =>
```
[the remark this answers is not in the document]
Comment (arempter) [reply src/main/scala/com/ing/wbaa/airlock/proxy/provider/LineageProviderAtlas.scala]: yes indeed, I will add this

File: src/main/scala/com/ing/wbaa/airlock/proxy/provider/atlas/LineageHelpers.scala [hunk 2]
```diff
@@ -191,30 +134,40 @@ trait LineageHelpers extends LazyLogging with RestClient {
     val bucketNameFromCopySrc = lh.copySource.get.split("/").head
```
[the remark this answers is not in the document]
Comment (arempter) [reply src/main/scala/com/ing/wbaa/airlock/proxy/provider/atlas/LineageHelpers.scala]: true, it is only called though if lineageHeaders.copySource.getOrElse("").length > 0. But I will update for clarity

File: src/main/scala/com/ing/wbaa/airlock/proxy/api/PostRequestActions.scala [hunk 3]
```diff
@@ -28,7 +28,7 @@ trait PostRequestActions {
   protected[this] def setDefaultBucketAcl(bucketName: String): Future[Unit]
 
   private[this] def createAtlasLineage(response: HttpResponse, httpRequest: HttpRequest, userSTS: User, clientIPAddress: RemoteAddress)(implicit id: RequestId): Future[Done] =
-    if (atlasSettings.atlasEnabled && (response.status == StatusCodes.OK || response.status == StatusCodes.NoContent)) {
+    if (atlasSettings.atlasEnabled && kafkaSettings.kafkaEnabled && (response.status == StatusCodes.OK || response.status == StatusCodes.NoContent)) {
```
Comment (kr7ysztof) [inline src/main/scala/com/ing/wbaa/airlock/proxy/api/PostRequestActions.scala]: does it mean to write to atlas we need to enable bucket notification?

```scala
private[this] def createBucketNotification(response: HttpResponse, httpRequest: HttpRequest, s3Request: S3Request,
      userSTS: User)(implicit id: RequestId): Future[Done] =
    httpRequest.method match {
      case HttpMethods.POST | HttpMethods.PUT | HttpMethods.DELETE if **kafkaSettings.kafkaEnabled** ..... 
```

File: docker-compose.yml [hunk 4]
```diff
@@ -55,9 +55,11 @@ services:
     - "8080:8080"
 
   atlas:
-    image: wbaa/airlock-dev-apache-atlas:0.0.5
+    image: wbaa/airlock-dev-apache-atlas:0.0.6
```
Comment (kr7ysztof) [inline docker-compose.yml]: we need separate kafka and zookeeper to test locally so let's go back to 0.0.5

**Pull request merged**